### PR TITLE
Direct kc.home.dir to Maven's target folder

### LINKED ISSUE #12850: Quarkus developer setup creates files in user's home directory
### Description

When starting Quarkus for development as explained in the README.md

```
mvn -f server/pom.xml compile quarkus:dev -Dquarkus.args="start-dev"
```

this will setup the Keycloak Home directory in the user's home directory. This is unexpected to the user, as those are not cleaned from that folder when running `mvn clean`.

This then leads to errors like `Wrong user name or password [28000-197]` or others, which can only be analyzed when debugging them. 

This change suggests to have the data created in the target folder as it is temporary data. Users should be able to override it when they need to / understand the concept of `kc.home.dir`.

This also applied to `IDELauncher`.

### Discussion

_No response_

### Motivation

Improve the developer experience with Quarkus.

### Details

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
This prevents files created, for example, the store to end up in the user's home directory.

Closes #12850

### CODE REVIEW

**Comment by ahus1** [comment]
@DGuhr - I know the build is still about to run, maybe you'd still want to have a first look if it is going into the right direction here. Thanks!

**Comment by ahus1** [comment]
Build is green, ready for review!

File: quarkus/server/pom.xml [hunk 1]
```diff
@@ -21,6 +21,12 @@
         </dependency>
     </dependencies>
 
+    <properties>
+        <!-- define this as a property, so it can still be overwritten on the CLI -->
+        <!-- direct the auto-created files to the target folder, so they are cleaned by "mvn clean" -->
+        <kc.home.dir>${project.build.directory}/kc</kc.home.dir>
```
[the remark this answers is not in the document]
Comment (DGuhr) [reply quarkus/server/pom.xml:27]: works for me, and definitely an improvement over randomly moving it to the users home dir, so approved :)

**Review by DGuhr — APPROVED** [review]

File: quarkus/server/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DGuhr) [reply quarkus/server/pom.xml:27]: just seen that I read this wrong and it definitely is where it belongs (and not as I misread it at project level). So even better ;)

**Pull request merged**